.github/workflows/rust.yml

```diff
@@ -33,6 +33,38 @@ jobs:
           components: rustfmt
       - run: cargo fmt --all -- --check
 
+  benchmark:
+    name: Performance regression check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@nightly
+      - name: Run benchmark
+        run: |
+          set -eo pipefail
+          cargo +nightly bench | tee output.txt
+      # Download previous benchmark result from cache (if exists)
+      - run: cat output.txt
+      - name: Download previous benchmark data
+        uses: actions/cache@v3
+        with:
+          path: ./cache
+          key: ${{ runner.os }}-benchmark
+      # Run `github-action-benchmark` action
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          # What benchmark tool the output.txt came from
+          tool: 'cargo'
+          # Where the output from the benchmark tool is stored
+          output-file-path: output.txt
+          # Where the previous data file is stored
+          external-data-json-path: ./cache/benchmark-data.json
+          # Workflow will fail when an alert happens
+          fail-on-alert: true
+          alert-threshold: "120%"
```

Comment:
I think it would be a good idea to "alert" every meaningful change, even if we only fail for some threshold. Especially I'd like to know if I (accidentially) improve performance with some change where I wouldn't expect it.

Reply:
This would be possible, if I give the applicaton an github acess token. so it can push the benchmark results to pr or github pages.

Reply:
I have remove the benchmark action for now. So the bench results can only be viewed at the logfile.